adsws/feedback/views.py

```diff
@@ -86,50 +99,128 @@ def prettify_post(post_data):
+                                        .format(post_data))
+                formatted_post_data = json.dumps(self.prettify_post(post_data))
+                current_app.logger.info('Data prettified: {0}'
+                                        .format(formatted_post_data))
+            except BadRequestKeyError as error:
+                current_app.logger.error('Missing keywords: {0}, {1}'
+                                         .format(error, post_data))
+                return err(ERROR_MISSING_KEYWORDS)
 
-        try:
-            current_app.logger.info('Prettifiying post data: {0}'
-                                    .format(post_data))
-            formatted_post_data = json.dumps(self.prettify_post(post_data))
-            current_app.logger.info('Data prettified: {0}'
-                                    .format(formatted_post_data))
-        except BadRequestKeyError as error:
-            current_app.logger.error('Missing keywords: {0}, {1}'
-                                     .format(error, post_data))
-            return err(ERROR_MISSING_KEYWORDS)
-
-        try:
-            slack_response = requests.post(
-                url=current_app.config['FEEDBACK_SLACK_END_POINT'],
-                data=formatted_post_data,
-                timeout=60
-            )
-        except (requests.exceptions.ConnectionError, requests.exceptions.Timeout):
-            return b'504 Gateway Timeout', 504
-        current_app.logger.info('slack response: {0}'
-                                .format(slack_response.status_code))
-
-        # Slack annoyingly redirects if you have the wrong end point
-        current_app.logger.info('Slack API' in slack_response.text)
-
-        if 'Slack API' in slack_response.text:
-            return err(ERROR_WRONG_ENDPOINT)
-        elif slack_response.status_code == 200:
-            return {}, 200
+            try:
+                slack_response = requests.post(
+                    url=current_app.config['FEEDBACK_SLACK_END_POINT'],
+                    data=formatted_post_data,
+                    timeout=60
+                )
+            except (requests.exceptions.ConnectionError, requests.exceptions.Timeout):
+                return b'504 Gateway Timeout', 504
+            current_app.logger.info('slack response: {0}'
+                                    .format(slack_response.status_code))
+
+            # Slack annoyingly redirects if you have the wrong end point
+            current_app.logger.info('Slack API' in slack_response.text)
+
+            if 'Slack API' in slack_response.text:
+                return err(ERROR_WRONG_ENDPOINT)
+            elif slack_response.status_code == 200:
+                return {}, 200
+            else:
+                return {'msg': 'Unknown error'}, slack_response.status_code
         else:
-            return {'msg': 'Unknown error'}, slack_response.status_code
+            return err(ERROR_UNKNOWN_ORIGIN)
+        return {}, 200
```

Comment:
We return a 200 even if emails fail to be sent, despite logging errors, if the user does not get any feedback that something went wrong, we risk to lose this feedback (nobody will be actively checking logs every day to make sure emails are being lost). I suggest to make email the first class citizen in this endpoint/class, if the email is not send (for forms or regular feedback), we return an error. Slack is secondary, if something doesn't reach slack but it reaches the email, we will notice and deal with it. So if slack fails, we log the error but the user request does not need to fail unless the email was also not sent.